src/main/java/edu/tamu/app/service/versioning/VersionOneService.java

```diff
@@ -1,71 +1,230 @@
+            Object status = card.getAttribute(statusNameAttribute).getValue();
+            Object cardType = ((IAssetType) card.getAttribute(assetTypeAttribute).getValue()).getToken();
+            Object[] owners = card.getAttribute(ownersAttribute).getValues();
+            cards.add(new Card(
+                number == null ? "" : number.toString(),
+                name == null ? "" : name.toString(),
+                description == null ? "" : description.toString(),
+                estimate == null ? "" : estimate.toString(),
+                getMembers(owners),
+                status == null ? "" : status.toString(),
+                cardType == null ? "" : cardType.toString()
+            ));
+        }
+
+        return cards;
     }
 
-    private String craftDataRequestUrl() {
-        return getUrl() + "/rest-1.v1/Data/Request";
+    @Override
+    public FeatureRequest push(FeatureRequest request) throws Exception {
+        Oid scopeId = services.getOid("Scope:" + request.getScopeId());
+        IAssetType requestType = services.getMeta().getAssetType("Request");
+        Asset newRequest = services.createNew(requestType, scopeId);
+        IAttributeDefinition nameAttribute = requestType.getAttributeDefinition("Name");
+        IAttributeDefinition descriptionAttribute = requestType.getAttributeDefinition("Description");
+        IAttributeDefinition scopeAttribute = requestType.getAttributeDefinition("Scope");
+
+        newRequest.setAttributeValue(nameAttribute, request.getTitle());
+        newRequest.setAttributeValue(descriptionAttribute, request.getDescription());
+        newRequest.setAttributeValue(scopeAttribute, "Scope:" + request.getScopeId());
+
+        services.save(newRequest);
+
+        return request;
+    }
+    
+    private List<Member> getMembers(Object[] owners) throws Exception {
+        List<Member> members = new ArrayList<Member>();
+        for (Object owner : owners) {
+            Oid memberId = services.getOid(owner.toString());
+            IAssetType memberType = services.getMeta().getAssetType("Member");
+            IAttributeDefinition nameAttribute = memberType.getAttributeDefinition("Name");
+            IAttributeDefinition avatarAttribute = memberType.getAttributeDefinition("Avatar");
+            
+            Query query = new Query(memberId);
+            query.getSelection().add(nameAttribute);
+            query.getSelection().add(avatarAttribute);
+            
+            QueryResult result = services.retrieve(query);
+            
+            Asset member = result.getAssets()[0];
+            String name = member.getAttribute(nameAttribute).getValue().toString();
+            String avatarId = member.getAttribute(avatarAttribute).getValue().toString();
+            members.add(new Member(name, getAvatarUrl(avatarId)));
+        }
+        return members;
+    }
+    
+    private String getAvatarUrl(String avatarId) throws Exception {
```

Comment:
We are going to have to rethink the avatar url. The url crafted from version one is only accessible with basic authentication.